expression/builtin_string.go

```diff
@@ -334,27 +339,59 @@ func (c *leftFunctionClass) getFunction(args []Expression, ctx context.Context)
 		return nil, errors.Trace(err)
 	}
 	bf.tp.Flen = args[0].GetType().Flen
+	if mysql.HasBinaryFlag(args[0].GetType().Flag) {
```

Comment:
1. we may use setBinFlagOrBinStr in getFunction.
2. we should not use binary flag to check whether a string is a binary string.